contracts/campaign/Campaign.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.7;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "./ICampaign.sol";
+import "./CampaignSecurityModule.sol";
+
+contract Campaign is ICampaign, CampaignSecurityModule {
+    using SafeMath for uint256;
+
+    uint256 public currentCampaign;
+    address public owner;
+
+    mapping(uint256 => ACampaign) public campaigns;
+
+    function initialize(
+        address _owner
+    ) public initializer {
+        __AccessControl_init();
+        __Pausable_init();
+        __ReentrancyGuard_init();
+
+        owner = _owner;
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _owner);
+        _setupRole(GOVERNANCE_ROLE, _owner);
+        _setupRole(CANCEL_ROLE, _owner);
+        _setupRole(PAUSE_ROLE, _owner);
+
+    }
+
+    function createCampaign(
+        uint256 _startShuttleNum,
+        uint256 _endShuttleNum,
+        uint256 _totalRewardAmount,
+        IERC20 _rewardToken
+    ) external onlyRole(GOVERNANCE_ROLE) {
+        uint256 totalShuttles_ = _endShuttle.sub(_startShuttle);
```

Comment:
are these shuttle number inclusive? 
If I want to run for shuttle 20-40, what arguments I will pass? 
19 -40 or 20-40 ?

Reply:
https://github.com/stakeall/cross-chain-staking-shuttle/blob/08f818f2cf07a0aecfc21dec9b46bc1c8a5d831f/contracts/campaign/Campaign.sol#L70

Shuttle Numbers are inclusive.